app/src/main/java/tech/ula/utils/ValidationUtility.kt

```diff
@@ -1,59 +1,53 @@
 package tech.ula.utils
 
-import android.content.res.Resources
 import tech.ula.R
 import java.util.regex.Matcher
 import java.util.regex.Pattern
 
-class ValidationUtility(systemResources: Resources) {
+class ValidationUtility(blacklisted_usernames: Array<String>) {
```

Comment:
Even though `blacklist_usernames` is being retrieved from XML which uses underscores, the parameter name should still probably be camelcase to follow Kotlin convention. It also might make more sense to send the list as a function parameter to `validateUsername` rather than as a class dependency for the `ValidationUtility`.